pkg/kind/kind.go

```diff
@@ -38,6 +45,50 @@ func SetUp() error {
 }
 
 func createKindCluster() error {
-	fmt.Println("TODO: Creating Kind cluster...")
+	fmt.Println("Creating Kind cluster...")
+
+	// Get kind config file
+	kindConfig, err := ioutil.TempFile(os.TempDir(), "kind-config-*.yaml")
+	if err != nil {
+		return fmt.Errorf("kind create: %w", err)
+	}
+
+	defer os.Remove(kindConfig.Name())
+
+	configRaw := "kind: Cluster\n" +
+		"apiVersion: kind.x-k8s.io/v1alpha4\n" +
+		"name: " + clusterName + "\n" +
+		"nodes:\n" +
+		"- role: control-plane\n" +
+		"  image: kindest/node:" + kubernetesVersion + "\n" +
+		"  extraPortMappings:\n" +
+		"  - containerPort: 31080\n" +
+		"    listenAddress: 127.0.0.1\n" +
+		"    hostPort: 80"
+	config := []byte(configRaw)
+
+	if _, err := kindConfig.Write(config); err != nil {
+		return fmt.Errorf("kind create: %w", err)
+	}
+
+	configFile := kindConfig.Name()
+	createCluster := exec.Command("kind", "create", "cluster", "--config", configFile)
+	if err := createCluster.Run(); err != nil {
+		return fmt.Errorf("kind create: %w", err)
+	}
+
+	if err := kindConfig.Close(); err != nil {
+		return fmt.Errorf("kind create: %w", err)
+	}
+
+	fmt.Println("    Waiting on cluster to be ready...")
+	time.Sleep(10 * time.Second)
```

Comment:
What is the sleep for here?

Reply:
From the [original script](https://github.com/csantanapr/knative-kind/blob/master/01-kind.sh#L49)

Reply:
Ok, but what is the sleep for here? :)  There's already a wait on the following line. Having said this the original script loops the wait (I dont really know why, tbh), so unlike the original script we only wait 1 second after the 10 seconds, meaning if the cluster takes longer than 11 seconds to come up, I think we'll error.

Reply:
Not sure, but FWIW taking a look at the `kube-system` on a kind cluster, it looks like the pods get launched in waves... first etcd, then apiserver / controller / scheduler, then finally coredns / kube-proxy / kindnet. 

(N=1) it took about 24 seconds for all kube-system pods to be running

Will clean this up

Reply:
I experimented a bit with this, and taking out the sleep statement caused the job to fail more often than not. It appears that there needs to be a non-zero period of time between when the cluster is created and running the `kubectl wait...` command. My assumption is this is because the apiserver pod is not always ready immediately after the `kind create cluster` command returns. Therefore, the `kubectl wait` command fails. By adding the sleep, I think we're ensuring that the apiserver is ready to field the wait command. (could the sleep be shorter? possibly, though 10s doesn't seem a terrible amount of time to wait... for reference, the end-to-end install takes under 2min on my machine.)

Reply:
Let's add a comment or something to help future readers out

Reply:
(And I still think we need to increase the timeout on the wait so it waits more than a second or add a loop)

Reply:
Since this PR has already closed, let's move the discussion over to #24

Reply:
> (And I still think we need to increase the timeout on the wait so it waits more than a second or add a loop)

I upped it to 30s in the new PR (wasn't counting on you to respond so quickly :) )